packages/perseus/src/widgets/interactive-graphs/utils.test.ts

```diff
@@ -119,4 +122,23 @@ describe("replaceOutsideTeX", () => {
             "\\text{Square }A\\text{ is }B\\text{ also}",
         );
     });
+
+    test("with a real $ inside a regular string", () => {
+        const string = "This sandwich is \\$12";
+        const convertedString = replaceOutsideTeX(string);
+
+        // For some reason, the parser starts a new text block once it hits
+        // the '$' character. The result is still the same though.
+        // This is functionally equal to "\\text{This sandwich is $12}"
+        expect(convertedString).toEqual(
+            "\\text{This sandwich is }\\text{$}\\text{12}",
+        );
+    });
+
+    test("with a real $ inside a TeX string", () => {
+        const mathString = "This sandwich is ${$}12$";
+        const convertedString = replaceOutsideTeX(mathString);
+
+        expect(convertedString).toEqual("\\text{This sandwich is }{$}12");
+    });
```

Comment:
How does this handle text with curly braces, backslashes or other TeX special characters in it? I think we might need to escape those:

```ts
    it("escapes curly braces", () => {
        const mathString = "Hello}{";
        const convertedString = replaceOutsideTeX(mathString);

        expect(convertedString).toEqual("\\text{Hello\\}\\{}");
    });

    it("escapes backslashes", () => {
        const mathString = "\\";
        const convertedString = replaceOutsideTeX(mathString);

        expect(convertedString).toEqual("\\text{\\\\}");
    });
```

Reply:
I think the parser assumes `{` and `}` to be part of TeX, so it separates it out before evaluating. It seems like the same thing with`\` and `^`. This results in some weird received strings.

For the second one, I think it only counts as one backslash, so the result would be `"\\text{\\}"`

Reply:
- `"Hello\\}\\{"` ==> `"\\text{Hello}\\text{}}\\text{{}"`
- `"\\"` ==> `"\\text{\\}"`

Reply:
I think both of those results are syntax errors in TeX:

- [example 1](https://www.khanacademy.org/devadmin/tex#JTVDdGV4dCU3QkhlbGxvJTdEJTVDdGV4dCU3QiU3RCU3RCU1Q3RleHQlN0IlN0IlN0Q=)
- [example 2](https://www.khanacademy.org/devadmin/tex#JTVDdGV4dCU3QiU1QyU3RA==)

We do need to escape the curly braces and backslashes to make these render correctly.